src/carousel/index.ts

```diff
@@ -0,0 +1,283 @@
+      this.temporaryNoTransition = true;
+      this.update();
+    };
+
+    if (rightBound >= threshold || this.currentSlide < 0) {
+      this.refs.track.addEventListener('transitionend', listener);
+    }
+  }
+
+  goToDot = (e: MouseEvent | TouchEvent) => {
+    e.preventDefault();
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+
+    const slide = parseInt((e.target as HTMLElement).getAttribute('data-index-to-go')) * slidesToShow;
+    this.goToSlide(slide);
+  }
+
+  getSlideStyle = () => {
+    const slideWidth = this.getSlideWidth();
+    return {
+      width: `${slideWidth}px`,
+      outline: 'none',
+    };
+  }
+
+  getTrackStyle = () => {
+    if (!(this.cloneItems() && this.props.items)) {
+      return;
+    }
+    const slideWidth = this.getSlideWidth();
+    const slideCount = this.cloneItems().length;
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const transition = typeof this.props.settings.transition === 'boolean' ? 
+      this.props.settings.transition : DEFAULT_SETTINGS.transition;
+    const speed = this.props.settings.speed || DEFAULT_SETTINGS.speed;
+
+    const trackWidth = (slideCount + 2 * slidesToShow) * slideWidth;
+    const pos = this.calcPos(this.currentSlide, slideWidth);
+    const tfm = `translate3d(-${pos}px, 0px, 0px)`;
+
+    const transformStyles = {
+      transform: tfm,
+      '-webkit-transform': tfm,
+      '-ms-transform': tfm,
+    };
+
+    const tsVal = speed + 'ms ' + 'ease';
+    const transitionStyles = transition === true ? {
+      '-webkit-transition': tsVal,
+      transition: tsVal
+    } : {};
+
+    const style = Object.assign({}, {
+      width: `${trackWidth}px`,
+    },
+      transformStyles,
+      transitionStyles);
+
+    if (this.temporaryNoTransition) {
+      delete style['transition'];
+      delete style['-webkit-transition'];
+      this.temporaryNoTransition = !this.temporaryNoTransition;
+    }
+    return style;
+  }
+
+  getDots = () => {
+    if (!this.props.items) {
+      return;
+    }
+    const slideCount = this.props.items.length;
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const slidesToScroll = this.props.settings.slidesToScroll || DEFAULT_SETTINGS.slidesToScroll;
+
+    const dotCount = Math.ceil((slideCount - slidesToShow) / slidesToScroll + 1);
+
+    return Array(dotCount).fill('dot');
+  }
+
+  getDotStyle = (i: number) => {
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const slidesToScroll = this.props.settings.slidesToScroll || DEFAULT_SETTINGS.slidesToScroll;
+
+    let leftBound = i * slidesToScroll;
+    let rightBound = i * slidesToScroll + slidesToShow - 1;
+
+    if (this.currentSlide >= leftBound && this.currentSlide <= rightBound) {
+      return { 'background-color': 'black' };
```

Comment:
Maybe add a css class instead of doing it this way? That way clients can more easily modify the style

Reply:
opps, I didn't see these comments until today. It was buried in the swamp of my code......Good thinking!